src/net/sourceforge/kolmafia/request/UseSkillRequest.java

```diff
@@ -730,34 +734,50 @@ private static boolean canSwitchToItem(final AdventureResult item) {
         && InventoryManager.hasItem(item, false);
   }
 
-  private static void equipForSkill(final AdventureResult item, final int skillId) {
-    if (!KoLCharacter.hasEquipped(item)) {
-      if (!InventoryManager.retrieveItem(item)) {
-        KoLmafia.updateDisplay(MafiaState.ERROR, "Cannot acquire " + item.getName() + ".");
+  private static void equipForSkill(final int skillId, final AdventureResult... items) {
```

Comment:
I would love to see a test for this function (or an appropriate one that calls it) if you have time